File: nysol_python/mcmd/methods/convTex/db/smp/mcombi.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mcombi'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
customer,item
A,a1
A,a2
A,a3
B,a4
B,a5
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
``customer`` 項目を単位に、 ``item`` 項目の2アイテムの組み合わせを求め、
``item1,item2`` という項目名で出力する。
``k=,f=`` で指定していない項目(ここでは ``item`` 項目)は、キーの最終行の値が出力される。
'''
script['sh_code']='''
mcombi k=customer f=item n=2 a=item1,item2 i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mcombi(k="customer", f="item", n="2", a="item1,item2", i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='基本例2'
script['comment']='''
``dup=True`` オプションを指定すると同一のアイテムの組み合せも出力される。
'''
script['sh_code']='''
mcombi k=customer f=item n=2 a=item1,item2 i=dat1.csv o=rsl2.csv -dup
'''
script['py_code']='''
nm.mcombi(k="customer", f="item", n="2", a="item1,item2", i="dat1.csv", o="rsl2.csv", dup=True).run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='順列を求める例'
script['comment']='''
``customer`` 項目を単位に、 ``item`` 項目の2アイテムの順列を求め、
``item1,item2`` という項目名で出力する。
'''
script['sh_code']='''
mcombi k=customer f=item n=2 a=item1,item2 -p i=dat1.csv o=rsl3.csv
'''
script['py_code']='''
nm.mcombi(k="customer", f="item", n="2", a="item1,item2", p=True, i="dat1.csv", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

script={}
script['title']='順列を求める例'
script['comment']='''
``item`` 項目を降順に並べ替えた後、
``item`` 項目の2アイテムの順列を求める。
'''
script['sh_code']='''
mcombi k=customer f=item n=2 s=item%r a=item1,item2 -p i=dat1.csv o=rsl4.csv
'''
script['py_code']='''
nm.mcombi(k="customer", f="item", n="2", s="item%r", a="item1,item2", p=True, i="dat1.csv", o="rsl4.csv").run()
'''
script['odatas']='rsl4.csv'
db['scripts'].append(script)

File: olddoc/mcmd/jp/examples/mcombi.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
customer,item
A,a1
A,a2
A,a3
B,a4
B,a5
EOF
)}

############## 例1
title="基本例"
comment=<<'EOF'
\verb|customer|項目を単位に、\verb|item|項目の2アイテムの組み合わせを求め、
\verb|item1,item2|という項目名で出力する。
\verb|k=,f=|で指定していない項目(ここでは\verb|item|項目)は、キーの最終行の値が出力される。
EOF
scp=<<'EOF'
more dat1.csv
mcombi k=customer f=item n=2 a=item1,item2 i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## 例2
title="基本例2"
comment=<<'EOF'
\verb|-dup|オプションを指定すると同一のアイテムの組み合せも出力される。
EOF
scp=<<'EOF'
mcombi k=customer f=item n=2 a=item1,item2 i=dat1.csv o=rsl2.csv -dup
more rsl2.csv
EOF
run(scp,title,comment)

############## 例3
title="順列を求める例"
comment=<<'EOF'
\verb|customer|項目を単位に、\verb|item|項目の2アイテムの順列を求め、
\verb|item1,item2|という項目名で出力する。
EOF
scp=<<'EOF'
mcombi k=customer f=item n=2 a=item1,item2 -p i=dat1.csv o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)

############## 例4
title="順列を求める例"
comment=<<'EOF'
\verb|item|項目を降順に並べ替えた後、
\verb|item|項目の2アイテムの順列を求める。
EOF
scp=<<'EOF'
mcombi k=customer f=item n=2 s=item%r a=item1,item2 -p i=dat1.csv o=rsl4.csv
more rsl4.csv
EOF
run(scp,title,comment)
